packages/overlay-sdk/src/account/account.ts

```diff
@@ -10,20 +10,36 @@ import { IntervalType, OverviewData } from "./types";
 
 export class OverlaySDKAccountDetails extends OverlaySDKModule {
   private sdk: OverlaySDK;
+  private overviewCache: Record<string, { data: any; lastUpdated: number }> = {};
 
   constructor(props: OverlaySDKCommonProps, sdk: OverlaySDK) {
     super(props);
     this.sdk = sdk;
   }
 
-  getOverview = async (interval: IntervalType = '1D', account?: Address, noCaching?: boolean) => {
+  getOverview = async (interval: IntervalType = '1D', account?: Address, refreshData?: boolean): Promise<OverviewData> => {
     let walletClient = account;
     if (!walletClient) {
       invariant(this.sdk.core.web3Provider, "Web3 provider is not set");
       walletClient = account ?? (await this.sdk.core.web3Provider?.requestAddresses())[0] as Address;
     }
     const chainId = this.core.chainId;
 
+    const cacheKey = `${walletClient}-${chainId}`;
+
+    let overviewData: OverviewData
+
+    if (this.overviewCache[cacheKey] && !refreshData) {
+      const cachedData = this.overviewCache[cacheKey];
+      const isCacheValid = Date.now() - cachedData.lastUpdated < 3 * 60 * 1000; // 3 minutes
+      if (isCacheValid) {
+        overviewData = cachedData.data;
+        return overviewData;
+      } else {
+        delete this.overviewCache[cacheKey];
+      }
+    }
+
     const [unwindPositions, liquidatedPositions, openPositions, numberOfPositions] = await Promise.all([
```

Comment:
In line 52 for transformOpenPositions should be added refreshData param, otherwise it will return cached data, not the updated

Reply:
nice catch